### ISSUE TITLE
Longer history for scores

### ISSUE THREAD

**Issue by davidwozabal** [boz1/tum-scopus-project#16]
In the statistics section, It should be possible to display longer histories for the scores. For example, it is easy to identify the number of FT50 publications for each author for each date (monthly/yearly) in the specified date range. Correspondingly, the tables and plots should cover the whole date range for the score parameter.

**Comment by boz1** [comment]
The monthly view covers only the months after Feb 19' since thats when we got the first data from the api, but on yearly view it is now possible to see all years in the date range selected. But still, we cannot provide any other data than scores, in this view, since we don't have the yearly distribution of citations, or other statistics. Here are some screenshots of the latest version of yearly view.
![Capture](https://user-images.githubusercontent.com/15712844/67165755-37939a00-f389-11e9-8c2e-c82da7c37270.PNG)
![Capture2](https://user-images.githubusercontent.com/15712844/67165756-37939a00-f389-11e9-82f4-0897c9c5c062.PNG)

**Comment by davidwozabal** [comment]
That already helps. However, I do not see why it should not be possible to compute, for example, H-indices on the fly.

What is wrong with the code snippet below? Where is the bottleneck here?

# calculation of h index for a single author a at a point in time t
# collect citations
for p1 in papers(a)
	counter(p1) = 0
	for p2 in papers_citing(p1)
		if p2.date <= t
			counter(p1) = counter(p1) + 1

# calculate H-index
counter = sort(counter, 'descending')
H = 0; i = 0
while i <= counter(i)
	H = H + 1
	i = i +1

**Comment by nicknn7** [comment]
For citations, we are already computing it this way. For historical h-index, you'd need a mapping citation -> date of citation. Potentially, you could compute this by considering the date of the paper citing each of the authors papers (similar to what you suggest above), but "on the fly" still seems problematic here. @boz1 and me will try to pre-compute this for each author.

**Comment by davidwozabal** [comment]
Yes, the publication date of the paper, is the date of the citation.

I would store all papers citing papers by authors in the authors list in a 'table' with columns: cited author, cited DOI, citing DOI, publication date of citing DOI.

Once this is stored, the code above should run without any problems.

**Comment by nicknn7** [comment]
This is subject to the API giving us a list of "cited by" papers. So far, it is not clear, whether we can query this.
